File: tests/fsharp/Reference.Tests.FSharp.NUnit/SerilogUnitTest.fs
namespace Reference.Tests.FSharp.NUnit

open NUnit.Framework
open Serilog
open System

type SerilogUnitTest () =
    let mutable _logger : ILogger = null

    [<SetUp>]
    member this.Init() =
        let mutable loggerConfig = new LoggerConfiguration()
        loggerConfig <- loggerConfig.MinimumLevel.Information()
        _logger <- loggerConfig.CreateLogger()
    
    [<Test>]
    member this.SerilogExample () =
        _logger.Verbose("Verbose in Serilog")
        _logger.Debug("Debug in Serilog")
        _logger.Information("Information in Serilog")
        _logger.Warning("Warning in Serilog")
        _logger.Error("Error in Serilog")
        _logger.Fatal("Fatal in Serilog")

        try
            raise (new Exception("Sample exception"))
        with 
            | ex -> _logger.Error(ex, "Exception error.")

        let logInfo = { Id = 6; Name = "Jacket"; Color = "Orange" }
        _logger.Information("Structured log: {@logStruct}", logInfo)

File: tests/csharp/Reference.Tests.CSharp.NUnit/SerilogUnitTest.cs
using System;
using NUnit.Framework;
using Serilog;

namespace Reference.Tests.CSharp.NUnit
{
    /// <summary>
    /// Serilog UnitTest
    /// </summary>
    public class SerilogUnitTest
    {
        private ILogger _logger;

        /// <summary>
        /// Logger configuration
        /// </summary>
        [SetUp]
        public void StartUp()
        {
            _logger = new LoggerConfiguration()
                .MinimumLevel.Information()
                .CreateLogger();
        }

        /// <summary>
        /// Serilog test
        /// </summary>
        [Test]
        public void SerilogExample()
        {
            _logger.Verbose("Verbose in Serilog");
            _logger.Debug("Debug in Serilog");
            _logger.Information("Information in Serilog");
            _logger.Warning("Warning in Serilog");
            _logger.Error("Error in Serilog");
            _logger.Fatal("Fatal in Serilog");

            try
            {
                throw new Exception("Sample exception");
            }
            catch(Exception ex)
            {
                _logger.Error(ex, "Exception error.");
            }

            _logger.Information("Structured log: {@logStruct}", new { Id = 6, Name = "Jacket", Color = "Orange" });
        }
    }
}